src/main/java/org/spongepowered/mod/mixin/core/event/player/MixinEventPlayerInteractBlock.java

```diff
@@ -79,4 +85,15 @@ public EntityInteractionType getInteractionType() {
         return Optional.fromNullable(new Cause(null, this.entityPlayer, null));
     }
 
+    @Override
+    public Optional<Vector3d> getClickedPosition() {
+        return Optional.of(VecHelper.toVector(this.pos).toDouble());
```

Comment:
The offset is where the crosshair on the client is pointing on the block. It's relative to the block in its 3D space, so x,y,z all range from 0 to 1. (Outside that range is invalid and should return absent). From some testing, when x = 0, that's the most west of the block, similarly with z = 0 that's the most north of the block, and y is the height where 0 is the bottom and 1 is the top.

This mixin is by no means useless, you just need to add an additional method on the event to be able to set the position before posted to the event bus.
The method `ItemInWorldManager.activateBlockOrUseItem` is invoked as a result of the packet, its last 3 parameters (`float p_180236_6_, float p_180236_7_, float p_180236_8_`) are the x y and z offset directly from the packet. It's possible to capture these parameters using an inject. The inject will be just before the packet is posted, so that you can call the new method on the event to set these values.